core/deployment/src/main/java/io/quarkus/deployment/dev/IDEDevModeMain.java

```diff
@@ -95,25 +96,24 @@ private void terminateIfRunning() {
         AppArtifactKey key = new AppArtifactKey(module.getArtifactCoords().getGroupId(),
                 module.getArtifactCoords().getArtifactId(), module.getArtifactCoords().getClassifier());
 
-        Set<String> sourceDirectories = new HashSet<>();
-        Set<String> sourceParents = new HashSet<>();
-        for (File srcDir : module.getSourceSourceSet().getSourceDirectories()) {
-            sourceDirectories.add(srcDir.getPath());
+        final Set<Path> sourceParents = new LinkedHashSet<>();
```

Comment:
Why not create PathsCollection directly here? Is it because it still allows duplicates?